frontend/src/app/core/services/dashboard.service.ts

```diff
@@ -1,66 +1,37 @@
 import { NewDashboard } from '@shared/models/dashboard/new-dashboard';
 import { UpdateDashboard } from '@shared/models/dashboard/update-dashboard';
+import { environment } from 'src/environments/environment';
+import { HttpInternalService } from './http-internal.service';
 
 @Injectable({ providedIn: 'root' })
 export class DashboardService {
-    //mock data
-    private dashboards: Dashboard[];
+    constructor(private http: HttpInternalService) { }
 
-    constructor() {
-        //mock data
-        this.dashboards = [
-            {
-                id: 1,
-                name: 'WebAPICharts',
-                icon: 'pi-chart-line'
-            },
-            {
-                id: 2,
-                name: 'JsCharts',
-                icon: 'pi-chart-line'
-            },
-            {
-                id: 3,
-                name: 'Python app',
-                icon: 'pi-chart-line'
-            },
-            {
-                id: 4,
-                name: 'ASP.NET Core',
-                icon: 'pi-chart-line'
-            },
-        ];
-    }
-
-    //mock methods
-    //TODO: Load real data from server
     public getIcons(): string[] {
         return ['pi-chart-bar', 'pi-chart-line'];
     }
 
-    public getAll(): Dashboard[] {
-        return this.dashboards;
+    public getAll(): Observable<HttpResponse<Dashboard[]>> {
+        return this.http.getFullRequest<Dashboard[]>(`${environment.coreUrl}/dashboard`);
     }
 
-    public get(id: string): Dashboard {
-        return this.dashboards.find(d => d.id === +id);
+    public get(id: string): Observable<HttpResponse<Dashboard>> {
+        return this.http.getFullRequest<Dashboard>(`${environment.coreUrl}/dashboard/${id}`);
     }
 
-    public addDashboard(newDashboard: NewDashboard): Dashboard {
-        this.dashboards.push({ id: this.dashboards.length + 1, name: newDashboard.name, icon: newDashboard.icon });
-        return this.dashboards[this.dashboards.length + 1];
+    public addDashboard(newDashboard: NewDashboard): Observable<HttpResponse<Dashboard>> {
+        return this.http.postFullRequest<Dashboard>(`${environment.coreUrl}/dashboard`, newDashboard);
     }
 
-    public deleteDashboard(id: number): boolean {
-        this.dashboards = this.dashboards.filter(d => d.id !== id);
-        return true;
+    public deleteDashboard(id: number) {
+        return this.http.deleteFullRequest<number>(`${environment.coreUrl}/dashboard/${id}`);
     }
 
-    public updateDashboard(updateDashboard: UpdateDashboard): Dashboard {
-        const key = this.dashboards.findIndex(el => el.id === updateDashboard.id);
-        this.dashboards[key] = updateDashboard;
-        return this.dashboards[key];
+    public updateDashboard(updateDashboard: UpdateDashboard): Observable<HttpResponse<Dashboard>> {
+        return this.http.putFullRequest<Dashboard>(`${environment.coreUrl}/dashboard/${updateDashboard.id}`, updateDashboard);
```

Comment:
You can send Id as part of the model. There is no need to send it as part of URL